packages/extension-ui/src/Popup/index.tsx

```diff
@@ -111,8 +112,10 @@ export default function Popup(): React.ReactElement {
   const history = useHistory();
 
   const _onAction = useCallback(
-    (to?: string): void => {
-      setWelcomeDone(window.localStorage.getItem('welcome_read') === 'ok');
+    async (to?: string): Promise<void> => {
+      const welcomeReadStatus = await localStorageStores.welcomeRead.get();
+
+      setWelcomeDone(welcomeReadStatus === 'ok');
```

Comment:
Yeah, it's a `setState` and thus a noop if the value remains unchanged, so this check on its own is not a problem. What does add some delay is waiting for local storage. But I did not put effort into optimizing it prematurely, since `onAction()` is not a high-frequency function.